component/local/estimator/estimator.go

```diff
@@ -0,0 +1,83 @@
+package estimator
+
+import (
+	"context"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "local.estimator",
+		Args:    Arguments{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+type Component struct {
+	opts            component.Options
+	metricsReceiver *prometheus.Interceptor
+	// This is maybe a hacky way to track unique combinations of seriesRef (map key), and label hash (map value).
+	// Could consume less memory by combining those two values into a single unique hash, but I'm not sure it's worth the additional effort ¯\_(ツ)_/¯
+	// Also, we may end up storing *actual* values for these, should we wish to report on them individually. I.E. There are x series with y label(s)
+	activeSeries map[storage.SeriesRef]map[uint64]struct{}
+}
+
+type Arguments struct {
+}
+
+type Exports struct {
+	metricsReceiver *prometheus.Interceptor `river:metrics_reciever,attr`
```

Comment:
```suggestion
	metricsReceiver *prometheus.Interceptor `river:metrics_receiver,attr`
```

Reply:
Spelling is hard 😂 

This is getting refactored anyhow, but, I have fixed the spelling :)